platform/linux-generic/pktio/tap.c

```diff
@@ -92,23 +92,30 @@ static int tap_pktio_open(odp_pktio_t id ODP_UNUSED,
 	int fd, skfd, flags;
 	uint32_t mtu;
 	struct ifreq ifr;
-	pktio_ops_tap_data_t *tap = odp_ops_data(pktio_entry, tap);
+	pktio_ops_tap_data_t *tap = NULL;
 
 	if (strncmp(devname, "tap:", 4) != 0)
 		return -1;
 
+	if (pool == ODP_POOL_INVALID)
```

Comment:
It probably belongs to its own patch, but now that you are at it, it could even be moved even further up, as it is probably faster than checking for "tap:" in the device string.

Reply:
Alloc/free vs. array has this disadvantage: you need to allocate the memory at some point and free it if the operation fails for any reason. It is better to delay the allocation until after some common checks.

stncmp: open calls are not on fast path ... no reason to optimize the performance ... but repeated memory alloc/free may affect some pool implementations

Reply:
Ok, I may be splitting hairs now, but I thought we were just checking whether the pool parameter passed to pktio_open was valid, and bail out if not. We are not actually using this pool to allocate this pktio's private data, right?

Reply:
No, this pool is used to allocate packets (for recv side).

Reply:
This is a parameter passed through from `odp_pktio_open()`. The spec simply says that this must be of type `ODP_POOL_PACKET`, so results are undefined if the caller doesn't abide by the spec. A courtesy validation check doesn't seem unreasonable, but any validation checking should be done in the main API, not in each individual driver. As such drivers should assume that `pool` is valid at entry since these entry points cannot be invoked directly by applications.

Reply:
I agree, this check should  be part of odp_pktio_open() or setup_pktio_entry() but this should be done in another PR. Here, I only move this check upper in the function (is not a new code).